Nag user when using IncrementalTaskInputs (#19202)
gradle/gradle: subprojects/docs/src/docs/userguide/migration/upgrading_version_7.adoc

[[changes_7.2]]
== Upgrading from 7.1 and earlier

[[incremental_task_inputs_deprecation]]
=== IncrementalTaskInputs type is deprecated

> The `IncrementalTaskInputs` type was used to implement _incremental tasks_, that is to say tasks which can be optimized to run on a subset of changed inputs instead of the whole input.

Reviewer: Very nit, but this is typographically preferrable:

```suggestion
The `IncrementalTaskInputs` type was used to implement _incremental tasks,_ that is to say tasks that can be optimized to run on a subset of changed inputs instead of the whole input.
```

> This type had a number of drawbacks, in particular the fact of not being able to determine to what input a change was associated with.

Reviewer: ```suggestion
This type had a number of drawbacks.
In particular using this type it was not possible to determine what input a change was associated with.
```
